Makefile

```
@@ -7,23 +7,28 @@
 CONFIG = config/application.yml
 HOST ?= localhost
 PORT ?= 3000
-GZIP_COMMAND ?= gzip
+
+.PHONY: brakeman check check_asset_strings docker_setup fast_setup fast_test help lint lint_country_dialing_codes lint_erb lint_optimized_assets lint_yaml lintfix normalize_yaml optimize_assets optimize_svg run run setup test update_pinpoint_supported_countries
+
+help: ## Show this help
+	@echo "--- Help ---"
+	@egrep -h '\s##\s' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}'
 
 all: check
 
-setup $(CONFIG): config/application.yml.default
+setup $(CONFIG): config/application.yml.default ## Runs setup scripts (updates packages, dependencies, databases, etc)
```

I might be at the limits of my Make-fu, but: Noticing that this is listed in the help output as `setup $(CONFIG)`. Could we have moved `$(CONFIG)` after the `:` (like with `test` below) to avoid `$(CONFIG)` being included in the output? Or updating the pattern used in the `awk` command of the `help` target?

---

this is saying `$(CONFIG) is an output`, which is correct, you can call `make config/application.yml` (the default value) and it will get created if it doesn't exist

---

Ah! TIL!

---

```suggestion
$(CONFIG) \
setup: config/application.yml.default ## Runs setup scripts (updates packages, dependencies, databases, etc)
```

I did just test this trick out and it works to fool the regex and preserve the old behavior, I'll make a PR

---

Based on how you described it, I could see some value in it being included in the help output if you can pass a file name to it (i.e. leaving it as-is). Up to you, though.

---

I don't think it's useful because it doesn't show us what `$(CONFIG)` is... but I got an idea and make a gnarly ruby script in https://github.com/18F/identity-idp/pull/5792